### PR TITLE
Fix artifact retrieval by specifying signed entity type

### LINKED ISSUE #3244: Client retrieves an artifact which is not of the expected type
## Why
We have noticed that the client does not return an error if it tries to access the hash of a signed entity which is created for another signed entity type.

## What

Return an error when trying to retrieve an artifact of the an unexpected signed entity type in the client. 

## How

- [x] Update the `SignedEntityStorer::get_signed_entity` function with a new `signed_entity_type_id` parameter
- [x] Update the calls to `SignedEntityStorer::get_signed_entity` in the `MithrilMessageService` implementation

### PR DESCRIPTION
**Pull request by turmelclem**
## Content
Fix an aggregator error, visible through `mithril-client-cli`, that occurs when attempting to fetch an existing snapshot by ID that is not of the expected signed entity type.

Signed entity records are now retrieved using both the signed entity ID and type. If no match is found, an empty result is returned instead of a deserialization error.

Example when trying to fetch a existing **cardano-transaction snapshot** with **cardano-block** command : 
```
./mithril-client cardano-block snapshot show d4889953d2dc795db1b732b5e01c7f7112374ff29b86f5c97b73131b0f1b14bb --unstable
```
before :
```
Error: Failed to get Cardano block snapshot with hash 'd4889953d2dc795db1b732b5e01c7f7112374ff29b86f5c97b73131b0f1b14bb'

Caused by:
    0: Internal error of the Aggregator
    1: internal server error: missing field `block_number_signed` at line 1 column 216
       
       Stack backtrace:
          0: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from

```
now : 
```
Error: Cardano blocks transactions snapshot not found for hash: '0909654ce0418597e9bf234f37961af207876634a3360fa259e1d770645ca94e'
```

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] All check jobs of the CI have succeeded
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)
  - [ ] Add ADR blog post or Dev ADR entry (if relevant)
  - [x] No new TODOs introduced

this PR closes #3244

### CODE REVIEW

File: mithril-aggregator/src/database/query/signed_entity/get_signed_entity.rs [hunk 1]
```diff
@@ -186,9 +193,12 @@ mod tests {
 
         let first_signed_entity_type = signed_entity_records[0].clone();
         let fetched_record = connection
```
Comment (Alenar) [inline mithril-aggregator/src/database/query/signed_entity/get_signed_entity.rs:195]: Can you add below (either in a new test or in this test) a case that ensure that `None` is returned when we retrieve an existing id but with the wrong type ?

**Review by Alenar — APPROVED** [review]
LGTM

**Review by jpraynaud — APPROVED** [review]
LGTM 👍
